components/style/properties/gecko.mako.rs

```diff
@@ -738,6 +745,24 @@ fn static_assert() {
     }
     ${impl_simple_copy('box_sizing', 'mBoxSizing')}
 
+    % for value in GRID_LINES:
+    pub fn set_${value.name}(&mut self, v: longhands::${value.name}::computed_value::T) {
```

Comment:
http://searchfox.org/mozilla-central/source/layout/style/nsRuleNode.cpp#8314 is what Gecko does. I see a couple of things missing:

 - The auto unit isn't handled here. Not sure if it needs to be; but there should be comments about how this is represented in gecko
 - Clamping between the maximum allowed unit isn't handled here.